File: 03/rb/03.rb
#!/usr/bin/env ruby

@input = open('input').readlines.map &:chomp

def go dx, dy = 1
    @input.each_slice(dy).map.with_index{|line, y| line[0][(y*dx) % line[0].size] == ?#}.count true
end

# part 1
p go 3

# part 2
p [1, 3, 5, 7, [1, 2]].map{|a| go *a}.inject :*

File: 03/jl/03.jl
#!/usr/bin/env julia

function go(input, dx, dy=1)
    sum(((y, line),) -> line[mod1(1 + dx*(y-1), length(line))] == '#', enumerate(input[1:dy:end]))
end

input = readlines("input")

println(go(input, 3))
println(prod(x -> go(input, x...), [1, 3, 5, 7, [1, 2]]))
